src/rgw/rgw_rest_s3.cc

```diff
@@ -5870,3 +5874,201 @@ bool rgw::auth::s3::S3AnonymousEngine::is_applicable(
 
   return route == AwsRoute::QUERY_STRING && version == AwsVersion::UNKNOWN;
 }
+
+
+const char *RGWS3Select::header_name_str[3] = {":event-type", ":content-type", ":message-type"};
+const char *RGWS3Select::header_value_str[3] = {"Records", "application/octet-stream", "event"};
+#include "s3select.h"
+
+RGWS3Select::RGWS3Select()
+{
+  set_get_data(true);
+  chunk_number = 0;
+  s3select_syntax = new s3select();
+  m_processed_bytes = 0;
+  m_previous_line = false;
+}
+
+RGWS3Select::~RGWS3Select()
+{
+  delete s3select_syntax;
```

Comment:
please use unique_ptr

Reply:
ok, 
the memory mng for s3-select , is about to change.
the idea is to reduce totally the amount of malloc/free per query (it could be enormous , thus impact RGW) 
i may use placement-new (or similar method), which means that unique_ptr will be obsolete.

Reply:
++